### PR TITLE
[PyTorch] Add ops for dropout and constant scale

### PR DESCRIPTION
**Pull request by timmoon10**
# Description

This PR adds fusible ops for dropout (heavily based on [`torch.nn.Dropout`](https://docs.pytorch.org/docs/stable/generated/torch.nn.Dropout.html)) and multiplying by a constant scalar. These are not performant, but they provide an API for future custom implementations or fusions.

## Type of change

- [ ] Documentation change (change only to the documentation, either a fix or a new content)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Infra/Build change
- [ ] Code refactoring

## Changes

- Add fusible ops for dropout and constant scale

# Checklist:

- [x] I have read and followed the [contributing guidelines](https://github.com/NVIDIA/TransformerEngine/blob/main/CONTRIBUTING.rst)
- [x] The functionality is complete
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: transformer_engine/pytorch/ops/basic/dropout.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        prev_op_grad_output_quantizer: Optional[Quantizer],
+        next_op_input_quantizer: Optional[Quantizer],
+    ) -> torch.Tensor:
+
+        # Compute dropout if training
+        out = input_
+        is_training = self.training
+        mask = None
+        if is_training:
+            keep_prob = 1 - self.dropout_probability
```
Comment (negvet) [inline transformer_engine/pytorch/ops/basic/dropout.py:45]: Handing this case similar to `torch.nn.Dropout`:

```suggestion
            if self.dropout_probability == 1:
                mask = torch.zeros_like(input_)
                out = mask
            else:
                keep_prob = 1 - self.dropout_probability
```

Comment (timmoon10) [reply transformer_engine/pytorch/ops/basic/dropout.py:45]: The existing impl should handle this case correctly. We will also replace this mask-based impl soon, so no need to optimize aggressively.
